src/games/strategy/triplea/delegate/BattleDelegateOrdered.java

```diff
@@ -0,0 +1,148 @@
+package games.strategy.triplea.delegate;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.data.CompositeChange;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Resource;
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.RouteScripted;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.delegate.AutoSave;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.engine.message.IRemote;
+import games.strategy.engine.random.IRandomStats.DiceType;
+import games.strategy.triplea.MapSupport;
+import games.strategy.triplea.TripleAUnit;
+import games.strategy.triplea.attachments.PlayerAttachment;
+import games.strategy.triplea.attachments.TerritoryAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.IBattle.BattleType;
+import games.strategy.triplea.delegate.IBattle.WhoWon;
+import games.strategy.triplea.delegate.dataObjects.BattleListing;
+import games.strategy.triplea.delegate.dataObjects.BattleRecord;
+import games.strategy.triplea.delegate.remote.IBattleDelegate;
+import games.strategy.triplea.formatter.MyFormatter;
+import games.strategy.triplea.oddsCalculator.ta.BattleResults;
+import games.strategy.triplea.player.ITripleAPlayer;
+import games.strategy.util.CompositeMatch;
+import games.strategy.util.CompositeMatchAnd;
+import games.strategy.util.CompositeMatchOr;
+import games.strategy.util.IntegerMap;
+import games.strategy.util.Match;
+import games.strategy.util.Tuple;
+
+public class BattleDelegateOrdered extends BattleDelegate {
+
+  /**
+   * Called before the delegate will run.
+   */
+  @Override
+  public void start() {
+    super.start();
+    // we may start multiple times due to loading after saving
+    // only initialize once
+
+    
+    Territory lastAmphib = null;
+    Territory lastNonAmphib = null;
+    Territory t;
+    IBattle battle;
+    Iterator<Territory> battleTerritories = m_battleTracker.getPendingBattleSites(true).iterator(); // get bombing/air raids
+    while( battleTerritories.hasNext() ) {
+      t = battleTerritories.next();
```

Comment:
You really should use a for-each loop here!
Instead of
```
Territory t;
Iterator<Territory> battleTerritories = m_battleTracker.getPendingBattleSites(true).iterator(); // get bombing/air raids
while(battleTerritories.hasNext()) {
  t = battleTerritories.next();
  //Code Here
}
```
You could just write
```
for(Territory t : m_battleTracker.getPendingBattleSites(true)) {// get bombing/air raids
  //Code Here
}
```